mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -1,17 +1,65 @@
-use crate::utxo::rpc_clients::UtxoRpcClientEnum;
 use crate::utxo::utxo_builder::{UtxoCoinBuildError, UtxoCoinBuilder, UtxoCoinBuilderCommonOps,
                                 UtxoFieldsWithHardwareWalletBuilder, UtxoFieldsWithIguanaPrivKeyBuilder};
 use crate::utxo::utxo_common::{block_header_utxo_loop, merge_utxo_loop};
 use crate::utxo::{GetUtxoListOps, UtxoArc, UtxoCommonOps, UtxoWeak};
 use crate::{PrivKeyBuildPolicy, UtxoActivationParams};
 use async_trait::async_trait;
 use common::executor::spawn;
-use common::log::info;
+use common::log::{info, LogOnError};
+use futures::channel::mpsc::Sender as AsyncSender;
 use futures::future::{abortable, AbortHandle};
 use mm2_core::mm_ctx::MmArc;
 use mm2_err_handle::prelude::*;
 use serde_json::Value as Json;
 
+pub enum UtxoSyncStatus {
+    SyncingBlockHeaders {
+        current_scanned_block: u64,
+        last_block: u64,
+    },
+    TemporaryError(String),
+    PermanentError(String),
+    Finished {
+        block_number: u64,
+    },
+}
+
+#[derive(Clone)]
+pub struct UtxoSyncStatusLoopHandle(AsyncSender<UtxoSyncStatus>);
```

Comment:
I think it's better to move `UtxoSyncStatusLoopHandle` declaration to `utxo_coin_builder.rs` since `utxo_arc_builder.rs` depends on `utxo_coin_builder.rs`, and `utxo_coin_builder.rs` shouldn't depend on `utxo_arc_builder.rs`.

Reply:
Done